static/app/components/searchSyntax/parser.tsx

```diff
@@ -896,10 +927,19 @@ export function parseSearch(
     ? {
         ...additionalConfig,
         ...Object.keys(defaultConfig).reduce((configAccumulator, key) => {
-          configAccumulator[key] =
-            typeof defaultConfig[key] === 'object'
-              ? new Set([...defaultConfig[key], ...(additionalConfig[key] ?? [])])
-              : defaultConfig[key];
+          if (typeof defaultConfig[key] === 'object') {
+            configAccumulator[key] = new Set([
+              ...defaultConfig[key],
+              ...(additionalConfig[key] ?? []),
+            ]);
+            return configAccumulator;
+          }
+          if (typeof defaultConfig[key] === 'boolean') {
+            configAccumulator[key] = additionalConfig[key] || defaultConfig[key];
+            return configAccumulator;
+          }
+
+          configAccumulator[key] = defaultConfig[key];
```

Comment:
This whole reducer is pretty nasty. I've simplified it here for clarity https://github.com/getsentry/sentry/pull/49720

Reply:
With the implementation I have in that PR you won't need to change anything in this

Reply:
yes... I didn't want to change it very much as I was not familiar with this code, but I am super glad that you improved it because it was difficult to read.